compiler/damlc/tests/daml-test-files/DamlDocHoogle.EXPECTED.hoogle

```diff
@@ -0,0 +1,57 @@
+
+module DamlDocHoogle
+
+-- | View docs
+
+data View
+
+
+View :: View
+
+
+getController :: I -> Party
+
+-- | T docs
+
+data T
+
+-- | T docs
+
+T :: Template T => Party -> T
+
+-- | TCall docs
+
+data TCall
+
+-- | TCall docs
+
+TCall :: Int -> TCall
+
+[i] :: TCall -> Int
+
+-- | TCall docs
+
+TCall_Choice :: Choice T => Int -> Update ()
+
+-- | I docs
+
+data I
+
+-- | I docs
+
+I :: Interface I => View -> I
```

Comment:
similar to `TCall_Choice`, this feels weird because `I` is not a data constructor. Perhaps to show the link with the viewtype we can have a specialized `view`?

```suggestion
view_I :: I -> View
```

maybe even

```suggestion
view @I :: I -> View
```

but I'm not sure how hoogle would handle the latter

Reply:
> The focus really of this constructor was to show the existence of the interface

ah I see. hm maybe we can attach that to the line for the data declaration instead? but I 'm not sure if hoogle supports "stupid theta" constraints

```
data DamlInterface => I
```

this would match the data declaration we desugar to

Reply:
imo `I : Interface` would be confusing, since `I` is neither a value of type `Interface` nor a type of kind `Interface`

Reply:
again, I don't mind that too much if we're super clear to users that these are not "proper" type signatures, but I do wonder if we should be thinking about forking hoogle to support e.g. `template`, `choice` and `interface` at the same level as `data`

Reply:
I do think the best thing to do is fork hoogle and do it right, rename it to doogle, have a good time. But theres the additional maintenance there

Reply:
Alternative DAHoogle

Reply:
DAhoo ;)

Reply:
Similar to the Choice approach, we could have the interface just be
`data InterfaceWithView View => I`